### ISSUE TITLE
multi-pattern SEGV

### ISSUE THREAD

**Issue by rurban** [rurban/re-engine-PCRE2#26]
```
perl -Dr -Mblib -Mre::engine::PCRE2 -e'
${^RE_DEBUG_FLAGS}=0xff;
my $re1 = qr/foo/;
my $re2 = qr/bar/;
"foobar" =~ /$re1$re2/'
```
PCRE2 op_comp: multi pattern 2. fallback to core
Segmentation fault: 11
in Assembling pattern from 2 elements S_concat_pat()
with at least one part of the pattern being a plugin pattern, not a core pattern.

see PR #25 

the single pattern variant works fine, it doesn't need to fall-back. but this needs to check the
plugin source of the pattern. It is only valid for a PCRE2 pattern.
```
${^RE_DEBUG_FLAGS}=0xff;
my $re1 = qr/foo/;
"foobar" =~ /$re1/'
```

e.g.:
```
${^RE_DEBUG_FLAGS}=0xff;
my $re1;
{ use re::engine::RE2; $re1 = qr/foo/; }
"foobar" =~ /$re1/'
```

**Comment by rurban** [comment]
Fixed with f18499cef18724e1dc4a0ec5604c7c2820737bd4 (mostly)

**Comment by mhaas** [comment]
@rurban Amazing, thanks!
